### PR TITLE
Pagination component

### LINKED ISSUE #253: Create pagination component
#### Description

We will need a pagination component as seen here https://www.figma.com/file/rBjfm2tJ2Pr9M0UNTduU9v/Joystream-Atlas?node-id=3198%3A4911

#### Requirements

- `Pagination ` component added to `src/shared/components`
- Component added to the Storybook

### PR DESCRIPTION
**Pull request by drillprop**
Fix #253

### CODE REVIEW

File: src/shared/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    <PaginationWrapper>
+      {startingIdx > 1 && (
+        <StyledLink isChevron={true} to={`${url}/${prevPage}`}>
+          <Icon name="chevron-left" />
+        </StyledLink>
+      )}
+      {Array.from({ length: totalPages })
+        .map((_, i) => (
+          <StyledLink key={i + startingIdx} to={`${url}/${i + startingIdx}`}>
+            {i + startingIdx}
```
Comment (mikkio-j) [inline src/shared/components/Pagination/Pagination.tsx]: First displayed page is 0 - I know it feels natural but I believe we should display 1 as a first page in the component

File: src/shared/components/Pagination/Pagination.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { Meta, Story } from '@storybook/react'
+import React from 'react'
+import { MemoryRouter, Route, useLocation } from 'react-router-dom'
```
Comment (mikkio-j) [inline src/shared/components/Pagination/Pagination.stories.tsx]:
```suggestion
import { MemoryRouter } from 'react-router-dom'
```

File: src/shared/components/Pagination/Pagination.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import Pagination from './Pagination'
+
+const Router: React.FC = (children) => {
```
Comment (mikkio-j) [inline src/shared/components/Pagination/Pagination.stories.tsx]: We don't need that since you are using decorator

**Review by mikkio-j — APPROVED** [review]

**Review by kdembler — CHANGES_REQUESTED** [review]
I'll get to the code later, looked at the storybook so far:
1) I'm not sure we should involve routing in page change. I don't see that as a benefit and it certainly complicates the data flow from this component to the consumer. I'd rather see that as a simple `value/onChange` component. What do you think?
2) The right arrow doesn't seem to work, whenever I click it, the 2nd page is chosen, regardless of what is currently selected
3) If I click the last element, the items should "scroll" to reveal the next items. For example, if there are 10 pages, 5 are visible at the beginning and I select the 5th page, the list should probably show pages 2-6 so I'm able to select the next one

File: src/shared/components/Pagination/Pagination.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { Meta, Story } from '@storybook/react'
+import React from 'react'
+import { MemoryRouter } from 'react-router-dom'
+import Pagination from './Pagination'
+
+export default {
+  title: 'Shared/Pagination',
+  component: Pagination,
+  argTypes: {
+    take: {
```
Comment (kdembler) [inline src/shared/components/Pagination/Pagination.stories.tsx]: This wasn't updated

File: src/shared/components/Pagination/Pagination.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react'
+import Icon from '../Icon'
+import { PaginationWrapper, PaginationButton } from './Pagination.style'
+
+type PaginationProps = {
+  itemsForPage?: number
```
Comment (kdembler) [inline src/shared/components/Pagination/Pagination.tsx]: Let's use `itemsPerPage` or `itemsOnPage`, `for` doesn't seem correct to me

File: src/shared/components/Pagination/Pagination.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+const Pagination: React.FC<PaginationProps> = ({
+  itemsPerPage = 0,
+  totalCount = 0,
+  currentPage = 1,
+  maxPaginationLinks = 5,
+  onClick,
+}) => {
+  const [page, setPage] = useState(currentPage)
```
Comment (kdembler) [inline src/shared/components/Pagination/Pagination.tsx]: Let's just rely on value from outside, we probably don't need internal state here

File: src/shared/components/Pagination/Pagination.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  totalCount?: number
+  currentPage?: number
+  maxPaginationLinks?: number
+  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+}
+
+const Pagination: React.FC<PaginationProps> = ({
+  itemsPerPage = 0,
+  totalCount = 0,
+  currentPage = 1,
```
Comment (kdembler) [inline src/shared/components/Pagination/Pagination.tsx]: Let's use 1-based indexing only for display, the values should be 0-based

File: src/shared/components/Pagination/Pagination.tsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react'
+import Icon from '../Icon'
+import { PaginationWrapper, PaginationButton } from './Pagination.style'
+
+type PaginationProps = {
+  itemsPerPage?: number
+  totalCount?: number
+  currentPage?: number
+  maxPaginationLinks?: number
+  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
```
Comment (kdembler) [inline src/shared/components/Pagination/Pagination.tsx]: Let's make this `onChange` / `onPageChange` and pass the new page number to the consumer

**Review by kdembler — APPROVED** [review]

**Pull request merged**